akka-actor/src/main/scala/akka/io/SelectionHandler.scala

```diff
@@ -159,6 +166,14 @@ private[io] object SelectionHandler {
             channelActor ! new ChannelRegistration {
               def enableInterest(ops: Int): Unit = enableInterestOps(key, ops)
               def disableInterest(ops: Int): Unit = disableInterestOps(key, ops)
+              def cancel(): Unit = {
+                key.cancel()
+                // On Windows the selector does not effectively cancel the registration until after the
+                // selector has woken up. Because here the registration is explicitly cancelled, we wake up the
+                // selector to make sure the cancellation (e.g. sending a RST packet for a cancelled TCP connection)
+                // is performed immediately.
+                selector.wakeup()
+              }
```

Comment:
The other callbacks here are wrapped in a `Task`. I'm not sure I quite understand the requirements/invariants here, should the implementation of `cancel()` also be wrapped like that or is that not necessary here?